helix-term/src/commands.rs

```diff
@@ -4839,7 +4852,7 @@ fn find_next_char_until_newline<M: CharMatcher>(
 }
 
 /// Decrement object under cursor by `amount`.
-fn increment_impl(cx: &mut Context, amount: i64) {
+fn increment_impl(cx: &mut Context, amount: i64, increase_by: i64) {
```

Comment:
Instead of passing amounts to `increase_impl`, we can create an enum

```rust
enum IncrementDirection {
    Increase,
    Decrease,
}
```

And pass that to `increment_impl`, then check the `cx.count` and `cx.register` in the body of `increment_impl` and then decide whether to use a positive or negative amount based on the passed direction.

Reply:
Okay, won't we need a 3rd value for the enum if we don't want to increase it at all?

> And pass that to increment_impl, then check the cx.count and cx.register in the body of increment_impl and then decide whether to use a positive or negative amount based on the passed direction.

Alright, that makes sense.

Reply:
Count can't be 0 so we don't need a third member